src/ui/common/src/components/workflows/artifact/metric/history.tsx

```diff
@@ -52,10 +52,31 @@ const MetricsHistory: React.FC<Props> = ({ historyWithLoadingStatus }) => {
         let timestamp = new Date(
```

Comment:
Does the following makes sense:
* take the max of `all_times` using `max(all_times.map(x => new Date(x)))`. We could figure out if there's any method in `Date` object to tell if the date is valid, and filter out any invalid one.
* render 'Unknown' if none timestamp above is available, and render the string version if the above block produce a valid date.
This should allow us to get rid of most branching in this block